docs/src/examples/redis-clustering-aware.md

```diff
@@ -0,0 +1,117 @@
+docker-compose -f docker-compose.yml up
+```
+
+Below we can see an example of a Redis node and it's Shotover sidecar. Notice they are running on the same network address (`172.16.1.2`) and the present directory is being mounted to allow Shotover to access the config and topology files.
+
+```YAML
+
+redis-node-0:
+  image: docker.io/bitnami/redis-cluster:6.0-debian-10
+  networks:
+  cluster_subnet:
+    ipv4_address: 172.16.1.2
+  volumes:
+    - redis-cluster_data-0:/bitnami/redis/data
+  environment:
+    - 'ALLOW_EMPTY_PASSWORD=yes'
+    - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2'
+
+shotover-0:
+  restart: always
+  depends_on:
+    - redis-node-0
+  image: shotover/shotover-proxy
+  network_mode: "service:redis-node-0"
+  volumes:
+    - type: bind
+      source: $PWD
+      target: /config
+
+```
+
+In this example we will use `redis-benchmark` with cluster mode enabled as our Redis cluster aware client application. 
+
+## Configuration
+
+First we will modify our `topology.yaml` file to have a single Redis source. This will:
+
+* Define how Shotover listens for incoming connection from our client application (`redis-benchmark`).
+* Configure Shotover to connect to the Redis node via our defined contact point.
+* Configure Shotover to rewrite all Redis ports with our Shotover port when the cluster aware driver is talking to the cluster, through Shotover.
+* Connect our Redis Source to our Redis cluster sink (transform).
+
+```yaml
+{{#include ../../../shotover-proxy/examples/redis-cluster-ports-rewrite/topology.yaml}}
+```
+
+Modify an existing `topology.yaml` or create a new one and place the above example as the files contents.
+
+You will also need a [config.yaml](https://raw.githubusercontent.com/shotover/shotover-proxy/main/shotover-proxy/config/config.yaml) to run Shotover.
+
+```console
+curl -L https://raw.githubusercontent.com/shotover/shotover-proxy/main/shotover-proxy/config/config.yaml --output config.yaml
+```
+
+## Starting
+
+We can now start the services with:
+
+```console
+docker-compose up -d
+```
+
+## Testing
+
+With everything now up and running, we can test out our client application. Let's start it up!
+
+First we will run `redis-benchmark` directly on our cluster. 
+
+```console
+redis-benchmark -h 172.16.1.2 -p 6379 -t set,get --cluster 
+```
+
+If everything works correctly you should see the following, along with the benchmark results which have been omitted for brevity. Notice all traffic is going through the Redis port on `6379`.
+
+
+```console
+Cluster has 3 master nodes:
+
+Master 0: d5eaf45804215f80cfb661928c1a84e1da7406a9 172.16.1.3:6379
+Master 1: d774cd063e430d34a71bceaab851d7744134e22f 172.16.1.2:6379
+Master 2: 04b301f1b165d81d5fb86e50312e9cc4898cbcce 172.16.1.4:6379
+```
+
+Now run it again but on the Shotover port this time.
+
+```console
+redis-benchmark -h 172.16.1.2 -p 6380 -t set,get --cluster 
```

Comment:
This step is failing for me 
![image](https://user-images.githubusercontent.com/5120858/143985509-508c5d5c-ba7b-4945-bdbb-e695ef51adcd.png)

Reply:
It works if I run it as `redis-benchmark -h 172.16.1.2 -p 6380 -t set,get --cluster`.
Maybe a bug in shotover redis MOVED handling?

Reply:
This step works fine for me

Reply:
Try running as:
```
redis-benchmark -h 172.16.1.2 -p 6380 -t set,get --cluster
redis-benchmark -h 172.16.1.3 -p 6380 -t set,get --cluster
redis-benchmark -h 172.16.1.4 -p 6380 -t set,get --cluster
```

On my machine there will always be one of these addresses that will fail.
Restarting the docker compose may result in a new ip being the one that fails.

So there is something like a 1/3 chance of the user failing to run shotover as documented.

Reply:
I managed to run into that issue you're describing. I went back to the 6 node cluster and it worked well. 

I think we can use the 3 leader/3 follower cluster as that is what the redis docs recommend. https://redis.io/topics/cluster-tutorial